tests/framework/randomness/test_crn.py

```diff
@@ -0,0 +1,301 @@
+        pytest.param(SequentialPopulation, True, cycle([2])),
+        pytest.param(SequentialPopulation, False, cycle([0])),
+        pytest.param(SequentialPopulation, False, cycle([2])),
+    ],
+)
+def test_multi_sim_basic_reproducibility_with_same_pop_growth(
+    pop_class,
+    with_crn,
+    sims_to_add,
+):
+    if with_crn:
+        configuration = {"randomness": {"key_columns": ["crn_attr1", "crn_attr2"]}}
+    else:
+        configuration = {}
+
+    sim1 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=sims_to_add)],
+        configuration=configuration,
+    )
+    sim2 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=sims_to_add)],
+        configuration=configuration,
+    )
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"])
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"])
+    assert_frame_equal(pop1, pop2)
+
+    for i in range(2):
+        sim1.step()
+        sim2.step()
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"])
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"])
+    assert_frame_equal(pop1, pop2)
+
+
+@pytest.mark.parametrize(
+    "pop_class, with_crn",
+    [
+        pytest.param(EntranceTimePopulation, True),
+        pytest.param(EntranceTimePopulation, False),
+        pytest.param(SequentialPopulation, True, marks=pytest.mark.xfail),
+        pytest.param(SequentialPopulation, False),
+    ],
+)
+def test_multi_sim_reproducibility_with_different_pop_growth(with_crn, pop_class):
+    if with_crn:
+        configuration = {"randomness": {"key_columns": ["crn_attr1", "crn_attr2"]}}
+    else:
+        configuration = {}
+
+    short, long = 1, 3
+    sim1 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=cycle([short, short]))],
+        configuration=configuration,
+    )
+    sim2 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=cycle([long, long]))],
+        configuration=configuration,
+    )
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"])
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"])
+    initial_pop_size = len(pop1)
+    assert_frame_equal(pop1, pop2)
+
+    time_steps = 7
+    for i in range(time_steps):
+        sim1.step()
+        sim2.step()
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"]).drop(columns="tracked")
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"]).drop(columns="tracked")
+
+    if with_crn:
+        overlap = pop1.index.intersection(pop2.index)
+        assert len(overlap) > initial_pop_size
+        assert_frame_equal(pop1.loc[overlap], pop2.loc[overlap])
+    else:
+        overlap = pop1.index[:initial_pop_size]
+        assert_frame_equal(pop1.loc[overlap], pop2.loc[overlap])
+
+
+class BrokenPopulation(BasePopulation):
```

Comment:
If you're proposing to add a bug-fix to support this use-case we probably should have a better name for this class.

Reply:
I'll roll it into the parameterization of the other population classes in this module with the fix.